worker/uniter/secrets/resolver.go

```diff
@@ -117,6 +125,49 @@ func (s *secretsResolver) rotateOp(
 	return &secretCompleter{op, opCompleted}, nil
 }
 
+func splitSecretChange(c string) (string, int) {
+	parts := strings.Split(c, "/")
+	if len(parts) < 2 {
+		return parts[0], 0
+	}
+	rev, _ := strconv.Atoi(parts[1])
+	return parts[0], rev
+}
+
+func (s *secretsResolver) expireOp(
+	_ resolver.LocalState,
+	remoteState remotestate.Snapshot,
+	opFactory operation.Factory,
+) (operation.Operation, error) {
+	if len(remoteState.ExpiredSecretRevisions) == 0 {
+		return nil, resolver.ErrNoOperation
+	}
+
+	// Nothing to do if we're no longer the leader.
+	if !remoteState.Leader {
+		return nil, resolver.ErrNoOperation
+	}
+
+	revSpec := remoteState.ExpiredSecretRevisions[0]
+	uri, rev := splitSecretChange(revSpec)
+	if rev == 0 {
+		s.logger.Warningf("ignoring invalid secret revision %q", revSpec)
```

Comment:
Do we want to return `resolver.ErrNoOperation` here?

Reply:
yup